src/transactions/EventManager.cpp

```diff
@@ -104,7 +105,7 @@ getPossibleMuxedData(SCAddress const& to, int64 amount, Memo const& memo)
 
     bool is_to_muxed_with_memo =
         has_memo && to.type() == SC_ADDRESS_TYPE_ACCOUNT;
-    if (!is_to_mux && !is_to_muxed_with_memo)
+    if (!allowMuxedIdOrMemo || (!is_to_mux && !is_to_muxed_with_memo))
```

Comment:
I realize I'm late to the review, and don't want to block the merge. So I'm just gonna comment here and address it in my followup.
I think it's better to not pass in `allowMuxedIdOrMemo`, instead check `getPossibleMuxedData` under an `if` condition outside. 
The early return condition is a little messy here, and the `allowMuxedIdOrMemo` is nothing but a short circuit, it's better to put it outside.

Reply:
If you had in on the outside, you'd duplicate the logic to create the amount SCVal right?

Reply:
yeah it'll call `makeI128SCVal`, personally I think that's better in conveying the intention. 
(sorry for borking your merge, forgot about the branch protection rule. )

Reply:
No worries. We feel free to make the change in your PR and we can take a look at what it looks like.